GVFS/GVFS.UnitTests/Service/RepoRegistryTests.cs

```diff
@@ -0,0 +1,200 @@
+using System.Text;
+
+namespace GVFS.UnitTests.Service
+{
+    [TestFixture]
+    public class RepoRegistryTests
+    {
+        [TestCase]
+        public void TryRegisterRepo_EmptyRegistry()
+        {
+            string dataLocation = Path.Combine("mock:", "registryDataFolder");
+
+            MockFileSystem fileSystem = new MockFileSystem(new MockDirectory(dataLocation, null, null));
+            RepoRegistry registry = new RepoRegistry(new MockTracer(), fileSystem, dataLocation, GVFSPlatform.Instance, new GVFSMountProcess());
+
+            string repoRoot = Path.Combine("c:", "test");
+            string ownerSID = Guid.NewGuid().ToString();
+
+            string errorMessage;
+            registry.TryRegisterRepo(repoRoot, ownerSID, out errorMessage).ShouldEqual(true);
+
+            Dictionary<string, RepoRegistration> verifiableRegistry = registry.ReadRegistry();
+            verifiableRegistry.Count.ShouldEqual(1);
+            this.VerifyRepo(verifiableRegistry[repoRoot], ownerSID, expectedIsActive: true);
+        }
+
+        [TestCase]
+        public void ReadRegistry_Upgrade_ExistingVersion1()
+        {
+            string dataLocation = Path.Combine("mock:", "registryDataFolder");
+            MockFileSystem fileSystem = new MockFileSystem(new MockDirectory(dataLocation, null, null));
+
+            string repo1 = Path.Combine("mock:", "code", "repo1");
+            string repo2 = Path.Combine("mock:", "code", "repo2");
+
+            // Create a version 1 registry file
+            fileSystem.WriteAllText(
+                Path.Combine(dataLocation, RepoRegistry.RegistryName),
+$@"1
+{{""EnlistmentRoot"":""{repo1.Replace("\\", "\\\\")}"",""IsActive"":false}}
+{{""EnlistmentRoot"":""{repo2.Replace("\\", "\\\\")}"",""IsActive"":true}}
+");
+
+            RepoRegistry registry = new RepoRegistry(new MockTracer(), fileSystem, dataLocation, GVFSPlatform.Instance, new GVFSMountProcess());
+            registry.Upgrade();
+
+            Dictionary<string, RepoRegistration> repos = registry.ReadRegistry();
+            repos.Count.ShouldEqual(2);
+
+            this.VerifyRepo(repos[repo1], expectedOwnerSID: null, expectedIsActive: false);
+            this.VerifyRepo(repos[repo2], expectedOwnerSID: null, expectedIsActive: true);
+        }
+
+        [TestCase]
+        public void ReadRegistry_Upgrade_NoRegistry()
+        {
+            string dataLocation = Path.Combine("mock:", "registryDataFolder");
+            MockFileSystem fileSystem = new MockFileSystem(new MockDirectory(dataLocation, null, null));
+            RepoRegistry registry = new RepoRegistry(new MockTracer(), fileSystem, dataLocation, GVFSPlatform.Instance, new GVFSMountProcess());
+            registry.Upgrade();
+
+            Dictionary<string, RepoRegistration> repos = registry.ReadRegistry();
+            repos.Count.ShouldEqual(0);
+        }
+
+        [TestCase]
+        public void TryGetActiveRepos_BeforeAndAfterActivateAndDeactivate()

```

Comment:
Can we break this test and `TryDeactivateRepo` up so that they are not doing so much and if something fails other tests will run?  This makes it easier to know what is failing as well as seeing what the test is testing.

Reply:
As discussed this is a pre-existing test class. We can discuss and create a follow-up PR if required.